File: Patterns/9. Diamond Patterns/88. Full Diamond Alphabets/fda.py
n = int(input("Enter a Number : "))
row=1

for i in range(1,2*n):
    if(i<=n):
        row=i
    else:
        row=2*n-i
    
    # print spaces
    for j in range(1,n-row+1):
        print(end='  ')
    # print first half
    for j in range(1,row+1):
        print(chr(ord('A')+j-1), end=' ')
    # print second half
    for j in range(row-1,0,-1):
        print(chr(ord('A')+j-1), end=' ')
    print()

'''
Output:

        A 
      A B A 
    A B C B A 
  A B C D C B A 
A B C D E D C B A 
  A B C D C B A 
    A B C B A 
      A B A 
        A
         
'''

File: Patterns/09. Diamond Patterns/88. Full Diamond Alphabets/fda.java
import java.util.*;
public class fda{
    public static void main(String [] args){
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter a Number : ");
        int n = sc.nextInt();

        int row=1;

        for(int i=1; i<=2*n-1; i++){
            if(i<=n){
                row = i;
            }else{
                row = 2*n-i;
            }
            // print spaces
            for(int j=1; j<=n-row; j++){
                System.out.print("  ");
            }
            // print first half
            for(int j=1; j<=row; j++){
                System.out.print((char)('A'+j-1)+ " ");
            }
            // print second half
            for(int j=row-1; j>=1; j--){
                System.out.print((char)('A'+j-1) + " ");
            }
            System.out.println();
        }
        
    }
}

/*
Output

        A 
      A B A 
    A B C B A 
  A B C D C B A 
A B C D E D C B A 
  A B C D C B A 
    A B C B A 
      A B A 
        A

*/